file_browser/src/components/AddDIalog.tsx

```diff
@@ -0,0 +1,112 @@
+import { useState } from "react";
+import { SystemItem } from "../classes/SystemItem";
+import { FileClass } from "../classes/File";
+import { Folder } from "../classes/Folder";
+import { observer } from "mobx-react-lite";
+import fileSystemStore from "../stores/FileSystemStore";
+
+const options = ["File", "Folder"];
+
+const AddDialog = observer(({
+  onClose
+}: {
+  onClose: (state: boolean) => void;
+}) => {
+
+  enum Selected {
+    FileOption = 0,
+    FolderOption = 1
+  }
+
+  const [selected, setSelected] = useState<number>(Selected.FileOption);
+  const [name, setName] = useState("");
+  const [content, setContent] = useState("");
+  const [error, setError] = useState("");
+
+  const handleAdd = (item: SystemItem) => {
+    (fileSystemStore.selectedItem as Folder).add(item);
+  };
+
+  const onChange = (i: any) => {
+    setSelected((prev: any) => (i === prev ? null : i));
+    setError("")
+  }
+
+  const createNewItem = () => {
+    let item: SystemItem;
+    /// folder
+    if (selected === Selected.FolderOption) {
+      if (name !== "") {
+        item = new Folder(name, fileSystemStore.selectedItem as Folder);
+        handleAdd(item);
+        onClose(false);
+      } else setError("You must choose a name for the folder");
+    }
+    //// file
+    else {
+      if (name !== "") {
+        item = new FileClass(
+          name,
+          fileSystemStore.selectedItem as Folder,
+          content
+        );
+        handleAdd(item);
+        onClose(false);
+      } else setError("You must choose name for the file");
+    }
+  }
+
+  return (
+    <dialog className="dialog" open>
+      {options.map((option, index) => (
+        <label key={index}>
+          {option}
+          <input
+            type="checkbox"
+            checked={index === selected}
+            onChange={() => onChange(index)}
+          />
+        </label>
+      ))}
```

Comment:
You can use the enum here as well:
`Object.values(Options).map(op => <...> {Options[op]} </...>)`
And this is how you make elegant readable code with enums.